File: app/src/main/java/com/veganbeauty/app/data/local/dao/CartDao.java
package com.veganbeauty.app.data.local.dao;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.veganbeauty.app.data.local.entities.CartItemEntity;

import java.util.List;
import kotlinx.coroutines.flow.Flow;

@Dao
public interface CartDao {
    @Query("SELECT * FROM cart_items")
    Flow<List<CartItemEntity>> getAllCartItems();

    @androidx.annotation.Nullable
    @Query("SELECT * FROM cart_items WHERE id = :itemId LIMIT 1")
    CartItemEntity getCartItemById(String itemId);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    long insertCartItem(CartItemEntity item);

    @Update
    int updateCartItem(CartItemEntity item);

    @Delete
    int deleteCartItem(CartItemEntity item);

    @Query("DELETE FROM cart_items")
    int clearCart();

    @Query("SELECT COUNT(*) FROM cart_items")
    int getCartItemCount();

    @Query("SELECT * FROM cart_items LIMIT 6")
    List<CartItemEntity> getCartItemsSync();
}

File: app/src/main/java/com/veganbeauty/app/data/local/dao/CartDao.kt
package com.veganbeauty.app.data.local.dao

import androidx.room.*
import com.veganbeauty.app.data.local.entities.CartItemEntity
import kotlinx.coroutines.flow.Flow

@Dao
interface CartDao {
    @Query("SELECT * FROM cart_items")
    fun getAllCartItems(): Flow<List<CartItemEntity>>

    @Query("SELECT * FROM cart_items WHERE id = :itemId LIMIT 1")
    suspend fun getCartItemById(itemId: String): CartItemEntity?

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertCartItem(item: CartItemEntity)

    @Update
    suspend fun updateCartItem(item: CartItemEntity)

    @Delete
    suspend fun deleteCartItem(item: CartItemEntity)

    @Query("DELETE FROM cart_items")
    suspend fun clearCart()
}
